src/client/Microsoft.Identity.Client/AppConfig/CiamAuthorityAdapter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Identity.Client
+{
+    internal class CiamAuthorityAdapter
+    {
+        public CiamAuthorityAdapter(string authority)
+        {
+            TransformAuthority(authority);
+        }
+
+        public CiamAuthorityAdapter(string instance, string tenant)
+        {
+            instance = instance.Trim().TrimEnd('/');
+            if (string.IsNullOrEmpty(tenant))
+            {
+                TransformAuthority(instance + '/' + tenant);
+            }
+            else
+            {
+                TransformedInstance = instance;
+                TransformedTenant = tenant;
+                TransformedAuthority = instance + '/' + tenant;
+            }
+        }
+
+        public string TransformedAuthority { get; private set; }
+
+        public string TransformedTenant { get; private set; }
+
+        public string TransformedInstance { get; private set; }
+
+        public string TransformedMetadata { get; private set; } = "{\"api-version\": \"1.1\",\"metadata\": [{\"preferred_network\": \"login.windows.net\",\"preferred_cache\": \"login.windows.net\",\"aliases\": [\"login.windows.net\",\"login.ciamlogin.com\"]}]}";
```

Comment:
Are `login.windows.net` and `login.ciamlogin.com` really the same thing?

Reply:
for the moment, but this will change.

Reply:
As of 3/24, login.window.net should be replaced by https://login.microsoftonline.com (this is the issuer for the v2.0 endpoint)